code/__HELPERS/icons.dm

```diff
@@ -126,4 +126,11 @@ proc/adjust_brightness(var/color, var/value)
 			colorsum[2] += RGB[2]
 			colorsum[3] += RGB[3]
 		final_average = rgb(colorsum[1]/total, colorsum[2]/total, colorsum[3]/total)
-	return final_average
\ No newline at end of file
+	return final_average
+
+/proc/empty_Y_space(var/icon/I) //Returns the amount of lines containing only transparent pixels in an icon, starting from the bottom

```

Comment:
sad that this was necessary, i think i remember helping with this months ago
was planning on putting together a graphics library with functions like this